plate_recognition.py

```diff
@@ -289,30 +442,22 @@ def main():
     for path in paths:
         if not path.exists():
             continue
-        with open(path, "rb") as fp:
-            api_res = recognition_api(
-                fp,
-                args.regions,
-                args.api_key,
-                args.sdk_url,
-                config=engine_config,
-                camera_id=args.camera_id,
-                mmc=args.mmc,
-            )
-
-        if (args.show_boxes or args.annotate_images) and "results" in api_res:
+        if Path(path).is_file():
+            if args.split_image:
+                results.append(process_split_image(path, args, engine_config))
+            else:
+                results.append(process_full_image(path, args, engine_config))
+        if args.show_boxes or args.annotate_images and "results" in results:
             image = Image.open(path)
-            annotated_image = draw_bb(image, api_res["results"], None, text_function)
+            annotated_image = draw_bb(image, results["results"], None, text_function)
```

Comment:
`results` is a list with one element per file. I believe we just want the last one.

```suggestion
            annotated_image = draw_bb(image, results["results"][-1], None, text_function)
```

Reply:
This causes errors Sir, tried to have this but the error is: TypeError: string indices must be integers

based on usage, draw_bb is a function that expects a list